datalad/plugin/export_archive.py

```diff
@@ -0,0 +1,126 @@
+    from datalad.support.annexrepo import AnnexRepo
+    from datalad.dochelpers import exc_str
+
+    import logging
+    lgr = logging.getLogger('datalad.plugin.export_archive')
+
+    repo = dataset.repo
+    committed_date = repo.get_committed_date()
+
+    # could be used later on to filter files by some criterion
+    def _filter_tarinfo(ti):
+        # Reset the date to match the one of the last commit, not from the
+        # filesystem since git doesn't track those at all
+        # TODO: use the date of the last commit when any particular
+        # file was changed -- would be the most kosher yoh thinks to the
+        # degree of our abilities
+        ti.mtime = committed_date
+        return ti
+    tar_args = dict(recursive=False, filter=_filter_tarinfo)
+
+    file_extension = '.{}{}'.format(
+        archivetype,
+        '{}{}'.format(
+            '.' if compression else '',
+            compression) if archivetype == 'tar' else '')
+
+    if filename is None:
+        filename = "datalad_{}".format(dataset.id)
+    if not filename.endswith(file_extension):
+        filename += file_extension
+
+    root = dataset.path
+    # use dir inside matching the output filename
+    # TODO: could be an option to the export plugin allowing empty value
+    # for no leading dir
+    leading_dir = file_basename(filename)
+
+    # workaround for inability to pass down the time stamp
+    with patch('time.time', return_value=committed_date), \
+            tarfile.open(filename, "w:{}".format(compression)) \
+            if archivetype == 'tar' \
+            else zipfile.ZipFile(
+                filename, 'w',
+                zipfile.ZIP_STORED if not compression else zipfile.ZIP_DEFLATED) \
+            as archive:
+        add_method = archive.add if archivetype == 'tar' else archive.write
+        repo_files = sorted(repo.get_indexed_files())
+        if isinstance(repo, AnnexRepo):
+            annexed = repo.is_under_annex(
+                repo_files, allow_quick=True, batch=True)
+        else:
+            annexed = [False] * len(repo_files)
+        for i, rpath in enumerate(repo_files):
+            fpath = opj(root, rpath)
+            if annexed[i]:
+                # resolve to possible link target
+                link_target = os.readlink(fpath)
+                if not isabs(link_target):
+                    link_target = normpath(opj(dirname(fpath), link_target))
+                fpath = link_target
+            # name in the archive
+            aname = normpath(opj(leading_dir, rpath))
+            try:
+                add_method(
+                    fpath,
+                    arcname=aname,
+                    **(tar_args if archivetype == 'tar' else {}))
+            except OSError as e:
+                if on_failure in('ignore', 'continue'):
+                    (lgr.warning if on_failure == 'continue' else lgr.debug)(
+                        'Skipped %s: %s',
+                        fpath, exc_str(e))
+
```

Comment:
So is that the way to generate archive from incomplete? On error continue?

If this plugin is used as a plugin for another action, might be impossible to provide specific control for this behavior without allowing all other errors also continue

Why not to make a dedicated option?

Reply:
I don't understand. There is a dedicated option for it. It specifically handles errors for individual files being added to the archive. it is distinct from any general error handler, to avoid this kind of confusion. So what do you want to be different?

Reply:
sorry, my bad -- I have missed that `on_error` is an option to the plugin and assumed that it was that global setting.
BTW -- would be nice if any test exercised this .
It seems that if I mistype `on_failure` value - it would be `ignore'.